examples/test_util/EclOutput.cpp

```diff
@@ -271,7 +271,7 @@ std::string EclOutput::make_real_string(float value) const
 
 std::string EclOutput::make_doub_string(double value) const
 {
-    char buffer [20];
+    char buffer [21];
```

Comment:
While this fix might be sufficient, it is bad to rely on sprintf(), which is a function that should be eradicated from the face of the earth. Take the opportunity to use snprintf() instead (and add a check for the return value)! Or something that directly works on strings, since there is significant overhead here anyway (concatenation might invoke reallocation for example). The function would also improve with a few comments explaining what is going on...

Reply:
> The function would also improve with a few comments explaining what is going on...

And lo and behold - the C++ streams interface could need something like a C++ style sprintf; but that is another matter.

Reply:
I agree with @atgeirr that this should use `snprintf`:
```
const std::size_t buffer_size = 20;
auto snprintf_value = snprintf(buffer, buffer_size, ...);
if (snprintf_value < 0 || snprintf_value >= buffer_size)
    ohhh - fuck();
```

Reply:
In C++17 the to_chars() and from_chars() functions promise to be good low-level tools for building fast and nice things that such conversions (guaranteed no allocations etc.) , but we are not there yet. Till then, snprintf() is the least bad choice...

Reply:
Alternative implementation:
```C++
#include <cmath>
#include <iostream>
#include <iomanip>
#include <sstream>
#include <string>

namespace { namespace Double {
    std::string format(double x)
    {
        std::ostringstream os;

        if (std::signbit(x)) {
            os << '-';
            x = -x;
        }
        else {
            os << ' ';
        }

        const auto exponent    = std::ceil(std::log10(x));
        const auto significand = x / std::pow(10.0, exponent);

        os << std::setw(15) << std::setprecision(14) << significand
           << ((exponent < 100.0) ? "D" : "")
           << std::showpos << std::internal
           << std::setw(3) << std::setfill('0')
           << static_cast<int>(exponent);

        return os.str();
    }
}}

int main() {
    const double value = -0.007441438752534199899921213;

    std::cout << Double::format(value) << '\n';
}
```
Prints:
```
$ ./a.out
-0.74414387525342D-02
```

This `Double::format()` generates a normed significand in the interval `[0.1 .. 1.0)`. Switching to `std::floor()` would produce a significand in the interval `[1.0 .. 10.0)`. When creating Fortran-compatible strings we should **probably** try to use normed significands. Ideally, there also would be no leading '0' character here.